src/Conn.h

```diff
@@ -90,8 +89,9 @@ class Connection : public BroObj {
 			// arguments for reproducing packets
 			const Packet *pkt);
 
-	HashKey* Key() const			{ return key; }
-	void ClearKey()				{ key = 0; }
+	ConnIDKey& Key()				{ return key; }
+	void ClearKey()					{ key_valid = false; }
+	bool IsKeyValid() const			{ return key_valid; }
```

Comment:
Might be helpful to document what these functions really mean.  "i.e. we call the key valid only when the connection is contained in the session map".